modules/core/src/main/java/org/apache/synapse/mediators/builtin/PropertyMediator.java

```diff
@@ -224,6 +223,15 @@ public boolean mediate(MessageContext synCtx) {
                         axis2smc.getAxis2MessageContext();
                 axis2MessageCtx.removeProperty(name);
 
+            } else if (XMLConfigConstants.SCOPE_CLIENT.equals(scope)
+                    && synCtx instanceof Axis2MessageContext) {
+
+                //Removing property from the Axis2-client Message Context
+                Axis2MessageContext axis2smc = (Axis2MessageContext) synCtx;
+                org.apache.axis2.context.MessageContext axis2MessageCtx =
+                        axis2smc.getAxis2MessageContext();
+                axis2MessageCtx.getOptions().setProperty(name, null);
```

Comment:
why the value is null

Reply:
We don't have a remove-property option at axis2MessageCtx.getOptions(). Therefore, in order to remove the property, I'm setting its value to null.

Reply:
can you add it as a code comment

Reply:
done